Anature/src/application/controllers/BattleController.java

```diff
@@ -1434,44 +1459,55 @@ private void playerTurn(BattleChoice choice, Runnable nextTurn)
 		}
 	}
 
-	private void enemyTurn(AiChoice enemyTurn, Runnable nextTurn)
+	private void enemyTurn(AiChoiceObject<?> enemyTurn, Runnable nextTurn)
 	{
-		Anature enemyAnature = mFightManager.getEnemyAnature();
-
-		switch(enemyTurn)
+		AiChoice aiChoice = enemyTurn.getAiChoice();
+		switch(aiChoice)
 		{
 			case Move1:
 				mClickQueue.enqueue(() ->
 				{
-					useAttack(enemyAnature, false, BattleChoice.Attack_1, 1);
+					useAttack(mFightManager.getEnemyAnature(), false, BattleChoice.Attack_1, 1);
 					activateAfterTurn(nextTurn);
 				}, "Enemy Attack 1");
 				break;
 
 			case Move2:
 				mClickQueue.enqueue(() ->
 				{
-					useAttack(enemyAnature, false, BattleChoice.Attack_2, 2);
+					useAttack(mFightManager.getEnemyAnature(), false, BattleChoice.Attack_2, 2);
 					activateAfterTurn(nextTurn);
 				}, "Enemy Attack 2");
 				break;
 
 			case Move3:
 				mClickQueue.enqueue(() ->
 				{
-					useAttack(enemyAnature, false, BattleChoice.Attack_3, 3);
+					useAttack(mFightManager.getEnemyAnature(), false, BattleChoice.Attack_3, 3);
 					activateAfterTurn(nextTurn);
 				}, "Enemy Attack 3");
 				break;
 
 			case Move4:
 				mClickQueue.enqueue(() ->
 				{
-					useAttack(enemyAnature, false, BattleChoice.Attack_4, 4);
+					useAttack(mFightManager.getEnemyAnature(), false, BattleChoice.Attack_4, 4);
 					activateAfterTurn(nextTurn);
 				}, "Enemy Attack 4");
 				break;
 
+			case Item_Consumed:
+				mClickQueue.enqueue(() ->
+				{
+					healthGain(mFightManager.itemUse(false, mFightManager.getEnemyIndex(), (HealthPotionBase) enemyTurn.getChoiceObject()), mEnemyHp);
+				}, "Enemy Item Use");
+
+			case Switch_Anature:
+				mClickQueue.enqueue(() ->
+				{
+					activateEnemySwitch(enemyTurn, nextTurn);
```

Comment:
You have to add:
activateAfterTurn(nextTurn);

After the switch call. Without it, the next turn will never happen.